obspy/core/stream.py

```diff
@@ -3118,6 +3119,85 @@ def remove_sensitivity(self, *args, **kwargs):
             tr.remove_sensitivity(*args, **kwargs)
         return self
 
+    def stack(self, group='all', type='linear', npts_tol=0):
+        """
+        Return stream with traces stacked by the same selected metadata.
+
+        :param group: Stack waveforms together which have the same metadata
+            given by this parameter. The parameter should name the
+            corresponding keys of the stats object,
+            e.g. `'{network}.{station}'` for stacking all
+            locations and channels of the stations and returning a stream
+            consisting of one stacked trace for each station.
+            This parameter can take two special values,
+            `'seedid'` which stacks the waveforms by seedid and
+            `'all'` (default) which stacks together all traces in the stream.
+        :param type: Type of stack, one of the following:
+            `'linear'`: average stack (default),
+            `('pw', order)`: phase weighted stack of given order,
+            see [Schimmel1997]_,
+            `('root', order)`: root stack of given order.
+        :param npts_tol: Tolerate traces with different number of points
+            with a difference up to this value. Surplus samples are discarded.
+
+        :returns: New stream object with stacked traces. The metadata of each
+            trace corresponds to the metadata of the original traces
+            if those are the same. Additionaly, the entries
+            `stack` (result of the format operation on the group parameter) and
+            `stack_count` (number of stacked traces)
+            are written to the stats object.
```

Comment:
maybe "stats object(s)" as the resulting stream could return more than one trace.